src/search/ran1.c

```
@@ -4,34 +4,65 @@ file contains function:
-}
\ No newline at end of file
+{
+    /* Initialize or reseed when idum is negative or state is zero */
+    if (*idum < 0 || ran1_state == 0) {
+        /* Use absolute value of idum as seed; ensure non-zero state */
+        uint64_t seed = (uint64_t)(*idum < 0 ? -(*idum) : *idum);
+        ran1_state = (seed == 0) ? UINT64_C(0x853c49e6748fea9b) : seed;
+        /* Warm up the generator */
+        (void)ran1_next64();
+        /* Convention: set idum positive after initialization */
+        if (*idum < 0) *idum = -(*idum);
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)